lib/auth/withAuth.ts

```diff
@@ -0,0 +1,67 @@
+import { headers } from "next/headers"
+
+import { hashToken } from "./hash-token"
+
+import { createClient } from "@/utils/supabase/server"
+
+export const withAuth = (
+  handler: (auth: {
+    request: Request
+    client: ReturnType<typeof createClient>
+    user: { id: string } | null
+    apiKey?: { mode: string; user_id: string } | null
+    timestamp: string
+  }) => void
+) => {
+  return async (request: Request) => {
+    const timestamp = new Date().toISOString()
+
+    const headerStore = headers()
+    const authHeader = headerStore.get("authorization")
+    const apiKey = authHeader ? authHeader.split(" ")[1] : ""
+    const hashedKey = await hashToken(apiKey)
```

Comment:
This is the same `withAuth` middleware as before. The diff is that it is using the hashed key instead.